### PR TITLE
Authentication modules

### PR DESCRIPTION
**Pull request by maartenba**

### CODE REVIEW

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+Back in the MyGet **Add authentication module** dialog:
+
+* Copy/paste the *Azure Active Directory application id* and specify the *Azure AD tenant* (e.g. `contoso.microsoft.com`).
+* (optional) If Azure Active Directory is hosted in Germany, China or any other environment from the default, enter the *Azure AD instance* (e.g.: `https://login.microsoftonline.com/`).
+* Click **Add** to save the configuration.
+
+## GitHub
+
+To allow users to login to MyGet using their GitHub account, enable the GitHub authentication module.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+Once the **Add authentication module** dialog is shown, we can find **Settings for GitHub**. Make sure to keep this window open as we will need it for the remainder of the configuration steps.
+
+In a separate browser window:
+
+* Log in to your GitHub account and **[register a new OAuth application with GitHub](https://github.com/settings/applications/new)**.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `Log in` --> `Sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+* Specify the general parameters to register a new OAuth application: *Application name*, *Homepage URL*, and an optional *Application description*.
+* In the *Authorization callback URL*, paste the callback URL that is displayed in MyGet *Settings for GitHub*.
+* Click the **Register application** button. GitHub will generate the credentials required to configure MyGet.
+
+Back in the MyGet **Add authentication module** dialog:
+
+* Copy/paste the *GitHub client id* and *GitHub client secret*.
+* Optionally, MyGet access can be limited to only allow certain GitHub organizations to login. Enter the organization names in the *Allowed GitHub organizations* field if desired. Organization names can be separated with semicolon. Leave the field blank to not enforce organization checks.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+* Click **Add** to save the configuration.
+
+## Google
+
+To allow users to login to MyGet using their Google account, enable the Google authentication module.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+Once the **Add authentication module** dialog is shown, we can find **Settings for Google**. Make sure to keep this window open as we will need it for the remainder of the configuration steps.
+
+In a separate browser window:
+
+* Log in to your Google account and open **[API credential management](https://console.developers.google.com/apis/credentials)**.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `Login` --> `Sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+* Click the **Create** button. Google will generate the credentials required to configure MyGet.
+
+Back in the MyGet **Add authentication module** dialog:
+
+* Copy/paste the *Google client id* and *Google client secret*.
+* Click **Add** to save the configuration.
+
+## Microsoft Account
+
+To allow users to login to MyGet using their Microsoft Account, enable the Microsoft Account authentication module.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+Once the **Add authentication module** dialog is shown, we can find **Settings for Microsoft Account**. Make sure to keep this window open as we will need it for the remainder of the configuration steps.
+
+In a separate browser window:
+
+* Log in to your Microsoft Account account and open **[My Applications](https://apps.dev.microsoft.com/)**.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `Log in` --> `Sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+* Under **Converged appications**, click the **Add an app** button.
+* In the *Application Name* field, enter an application name (e.g. *MyGet*).
+* Make sure *Let us help you get started*  is **disabled** and click **Create**.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: One whitespace character too much between `started* is`.

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+* Click the **Save** button. Microsoft Account will generate the credentials required to configure MyGet.
+
+Back in the MyGet **Add authentication module** dialog:
+
+* Copy/paste the *Microsoft Account application id* that is displayed under **[My Applications - Converged applications](https://apps.dev.microsoft.com/)**.
+* Click **Add** to save the configuration.
+
+## StackExchange
+
+To allow users to login to MyGet using their StackExchange account, enable the StackExchange authentication module.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+Once the **Add authentication module** dialog is shown, we can find **Settings for StackExchange**. Make sure to keep this window open as we will need it for the remainder of the configuration steps.
+
+In a separate browser window:
+
+* Log in to your StackExchange account and open **[Register a new app](https://stackapps.com/apps/oauth/register)**.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `Log in` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+* Click the **Register Your Application** button. StackExchange will generate the credentials required to configure MyGet.
+
+Back in the MyGet **Add authentication module** dialog:
+
+* Copy/paste the *StackExchange client id*, *StackExchange client secret* and *StackExchange key*.
+* Click **Add** to save the configuration.
+
+## WS-Federation (ADFS)
+
+To allow users to login to MyGet using their ADFS account, enable the WS-Federation authentication module.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: `login` --> `sign in`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+* In **Server Manager**, click **Tools**, and then select **AD FS Management**.
+* Under **Actions**, click **Add Relying Party Trust**.
+* On the *Welcome* page, choose **Claims aware** and click **Start**.
+* On the *Select Data Source* page, click **Enter data about the relying party manually**, and then click **Next**.
+* On the *Specify Display Name* page, enter a display name (e.g. *MyGet*). Optionally a description can be added as well. Click **Next**.
+* On the *Configure Certificate* page, click **Next**
+* On the *Configure URL* page, enable the* Enable support for the WS-Federation Passive protocol* checkbox. In the *Relying party WS-Federation Passive protocol URL* field, paste the reply URL that is displayed in MyGet *Settings for WS-Federation (ADFS)*. Click **Next**.
+* On the *Configure Identifiers page*, specify two identifiers for this relying party:
+  * The sign-on URL that is displayed in MyGet *Settings for WS-Federation (ADFS)*
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: It's `log on` or `sign in`, where does `sign-on` come from? :)

Comment (maartenba) [reply Docs.Site/Docs/Reference/Authentication-Modules.markdown]: From the form they see in front of them on MS website :-)

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown [hunk 13]
```diff
@@ -1,6 +1,6 @@
 # Authentication Modules
 
-The [MyGet Enterprise plan](https://www.myget.org/enterprise) provides support for external authentication modules to log in to the web application. This document describes the available authentication modules and how to configure them.
+The [MyGet Enterprise plan](https://www.myget.org/enterprise) provides support for external authentication modules to Sign in to the web application. This document describes the available authentication modules and how to configure them.
```
Comment (xavierdecoster) [inline Docs.Site/Docs/Reference/Authentication-Modules.markdown]: nit: lowercase `Sign`

File: Docs.Site/Docs/Reference/Authentication-Modules.markdown (hunk 8 above)
Comment (xavierdecoster) [reply Docs.Site/Docs/Reference/Authentication-Modules.markdown]: Missed this one?

**Pull request merged**